src/main/java/software/sigma/internship/scheduler/ScheduledUtil.java

```diff
@@ -0,0 +1,48 @@
+package software.sigma.internship.scheduler;
+
+import lombok.AllArgsConstructor;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.aggregation.AggregationResults;
+import org.springframework.data.mongodb.core.aggregation.MatchOperation;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import software.sigma.internship.dto.TestStatisticsDto;
+import software.sigma.internship.entity.TestStatistics;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.newAggregation;
+
+@Service
+@AllArgsConstructor
+public class ScheduledUtil {
+
+    private final MongoTemplate mongoTemplate;
+    private final KafkaTemplate<String, Map<String, List<TestStatisticsDto>>> kafkaTemplate;
+
+    @Scheduled(cron = "0 0 18 ? * *")
```

Comment:
It would be better to use properties for the schedule.

Reply:
agree